shaders/src/light_indirect.fs

```diff
@@ -170,6 +208,37 @@ vec3 getReflectedVector(const PixelParams pixel, const vec3 n) {
     return getSpecularDominantDirection(n, r, pixel.roughness);
 }
 
+// This function returns an IBL lookup direction, taking into account the current IBL type (e.g. infinite spherical, finite/local sphere/box).
+vec3 GetAdjustedReflectedDirection(const MaterialInputs material, const PixelParams pixel, const vec3 normal) {
+    if (frameUniforms.iblTechnique == IBL_TECHNIQUE_INFINITE) return getReflectedVector(pixel, normal);
+
+    // intersect the ray ray_pos + t * ray_dir with the finite geometry; done in the coordinate system of the finite geometry
+    vec3 ray_pos = getWorldPosition() + getWorldOffset() - frameUniforms.iblCenter;
+    vec3 ray_dir = getReflectedVector(pixel, normal);
+
+    vec3  r  = vec3(0.0); // resulting direction
+    float t0 = -1.0f;     // intersection parameter between ray and finite IBL geometry
+    
+    if (frameUniforms.iblTechnique == IBL_TECHNIQUE_FINITE_SPHERE) {
+        float R2 = frameUniforms.iblHalfExtents.r; // we store the squared radius to shave off a multiplication here
+        float A = 1.0; // in general, this should be dot(ray_dir, ray_dir) but we have just normalized it a couple of lines ago
+        float B = 2.0 * dot(ray_pos, ray_dir);
+        float C = dot(ray_pos, ray_pos) - R2;
+        vec2 roots = SolveQuadratic(A, B, C);
+        t0 = GetSmallestPositive(roots.x, roots.y);
+    }
+    else if (frameUniforms.iblTechnique == IBL_TECHNIQUE_FINITE_BOX) {
+        vec2 roots = IntersectAABB(ray_pos, ray_dir, -frameUniforms.iblHalfExtents, frameUniforms.iblHalfExtents);
+        t0 = GetSmallestPositive(roots.x, roots.y);
+    }
+
+    // translate results back to world space
+    vec3 intersection_point = ( t0 >= 0.0 ) ? ray_pos + t0 * ray_dir : getReflectedVector(pixel, shading_normal);
```

Comment:
Last parameter should be `normal` if I'm not mistaken.
```suggestion
    vec3 intersection_point = ( t0 >= 0.0 ) ? ray_pos + t0 * ray_dir : getReflectedVector(pixel, normal);
```
Alternatively, the very first step of this function could be compute the reflected vector, as it's needed in all possible paths, (result for infinite IBL, ray direction for intersection calculations, and fallback value here).

Reply:
Good catch, fixed in https://github.com/shapr3d/filament/pull/44/commits/2b1597ef3a384d9f250c8f016342ea93a469a947